examples/Thermostat/Device/ThermoBot.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) ZeroC, Inc.
+
+using IceRpc;
+using IceRpc.Features;
+using IceRpc.Slice;
+using Igloo;
+using System.Runtime.CompilerServices;
+
+namespace ThermostatDevice;
+
+/// <summary>Implements all the logic of our thermostat device, without any actual electronics or a real AC unit.
+/// </summary>
+internal class ThermoBot : Service, IThermoControlService
+{
+    internal Task ReadCompleted => _readTcs.Task;
+
+    private Stage _cooling;
+
+    // Protects all read-write fields.
+    private readonly object _mutex = new();
+
+    // Used to signal when the server has stopped reading.
+    private readonly TaskCompletionSource _readTcs = new(TaskCreationOptions.RunContinuationsAsynchronously);
+
+    private float _setPoint = 75.0F;
+    private float _temperature = 74.0F;
+
+    /// <inheritdoc />
+    public ValueTask ChangeSetPointAsync(
+        float setPoint,
+        IFeatureCollection features,
+        CancellationToken cancellationToken)
+    {
+        lock(_mutex)
+        {
+            if (setPoint < 65.0F)
+            {
+                throw new DispatchException(
+                    StatusCode.ApplicationError,
+                    "Invalid set point: the set point must be greater than or equal to 65°F.");
+            }
+            _setPoint = setPoint;
+            if (_temperature > setPoint)
+            {
+                _cooling = _temperature - setPoint > 5.0F ? Stage.Stage2 : Stage.Stage1;
+            }
+            else
+            {
+                _cooling = Stage.Off;
+            }
```

Comment:
I think this would look cleaner as a switch expression, instead of (what's effectively) nested ternary operators.
```suggestion
            _cooling = (_temperature - setPoint) switch {
			    > 5.0F => Stage.Stage2,
			    > 0.0F => Stage.Stage1,
			    _ => Stage.Off,
            };
```
Maybe this is too fancy though.

Reply:
> Maybe this is too fancy though.

I think it's too fancy. I think its more readable with the if statement.